applications/RomApplication/python_scripts/calculate_rom_basis_output_process.py

```diff
@@ -151,20 +152,27 @@ def ExecuteFinalize(self):
         rom_basis_dict["rom_settings"]["number_of_rom_dofs"] = numpy.shape(u)[1] #TODO: This is way misleading. I'd call it number_of_basis_modes or number_of_rom_modes
         rom_basis_dict["projection_strategy"] = "galerkin" # Galerkin: (Phi.T@K@Phi dq= Phi.T@b), LSPG = (K@Phi dq= b), Petrov-Galerkin = (Psi.T@K@Phi dq = Psi.T@b)
 
-        i = 0
-        for node in self.model_part.Nodes:
-            rom_basis_dict["nodal_modes"][node.Id] = u[i:i+n_nodal_unknowns].tolist()
-            i += n_nodal_unknowns
-
-        # Export the ROM basis dictionary
         if self.rom_basis_output_format == "json":
-            output_filename = self.rom_basis_output_name + "." + self.rom_basis_output_format
-            with open(output_filename, 'w') as f:
-                json.dump(rom_basis_dict, f, indent = 4)
+            # Storing modes in JSON format
+            i = 0
+            for node in self.model_part.Nodes:
+                rom_basis_dict["nodal_modes"][node.Id] = u[i:i+n_nodal_unknowns].tolist()
+                i += n_nodal_unknowns
+
+        elif self.rom_basis_output_format == "numpy":
+            # Storing modes in Numpy format
+            numpy.save('RightBasisMatrix.npy', u)
+            numpy.save('NodeIds.npy',  numpy.arange(1,((u.shape[0]+1)/n_nodal_unknowns), 1, dtype=int)   )
```

Comment:
I am curious about whether this would still work if there is a floating node within the model part, although this is highly unlikely. To provide an example, suppose there is a "FluidModelPart" that contains points that were generated from CAD and are not included in the computational model part.

Reply:
If that specific case does not work, it would not work for the JSON format either